### PR TITLE
move modular code development lesson

### PR DESCRIPTION
**Pull request by gcroci2**
See issue [#38](https://github.com/esciencecenter-digital-skills/good-practices-in-research-software-development/issues/38) (from the old repository for good-practices-in-research-software-development) for more details.

### CODE REVIEW

**Comment by gcroci2** [comment]
Ready to be reviewed :) @svenvanderburg @lyashevska

File: config.yaml [hunk 1]
```diff
@@ -62,6 +62,7 @@ episodes:
 - 2-documentation.md
 - 3-testing.md
 - 4-ci.md
+- 5-modular-code.md
```
Comment (svenvanderburg) [inline config.yaml:65]:
```suggestion
- 1-modular-code.md
- 2-documentation.md
- 3-testing.md
- 4-ci.md
```
I think this is usually the order that we teach it right? Can you also change the filename to `episodes/1-modular-code.md` accordingly?

File: episodes/5-modular-code.md [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Understand the key characteristics that make code modular and why it is advantageous.
+- Learn strategies for identifying potential modules within complex code to improve readability and reusability.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## What is modularity?
+
+Modularity refers to the practice of building software from smaller, self-contained, and independent elements. Each element is designed to handle a specific set of tasks, contributing to the overall functionality of the system.
```
Comment (svenvanderburg) [inline episodes/5-modular-code.md:23]: Maybe you can keep this part as an introduction, then mention that we will explain modular coding in the slides and link to the slides

**Review by svenvanderburg — CHANGES_REQUESTED** [review]
Great work @gcroci2. The main change here would be to refer to the slides instead of having the content of the slides duplicated here.

File: episodes/1-modular-code.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
+---
+title: Modular Code Development
+teaching: 30
+exercises: 20
```
Comment (svenvanderburg) [inline episodes/1-modular-code.md:4]:
```suggestion
exercises: 30
```
You also need time to discuss the solution (10 minutes)

**Review by svenvanderburg — APPROVED** [review]
Great, one more small comment. Then feel free to merge!

**Pull request merged**